hmpps-sqs-spring-boot-autoconfigure/src/test/kotlin/uk/gov/justice/hmpps/sqs/HmppsNoDlqQueueFactoryTest.kt

```diff
@@ -0,0 +1,385 @@
+    private val sqsClient = mock<AmazonSQS>()
+    private val snsClient = mock<AmazonSNS>()
+    private val topics = listOf(HmppsTopic(id = "sometopicid", arn = "some topic arn", snsClient = snsClient))
+    private lateinit var hmppsQueues: List<HmppsQueue>
+
+    @BeforeEach
+    fun `configure mocks and register queues`() {
+      whenever(sqsFactory.localStackSqsClient(anyString(), anyString(), anyString(), anyString(), anyBoolean()))
+        .thenReturn(sqsClient)
+      whenever(sqsClient.getQueueUrl(anyString())).thenReturn(GetQueueUrlResult().withQueueUrl("some queue url"))
+
+      hmppsQueues = hmppsQueueFactory.createHmppsQueues(hmppsSqsProperties, topics)
+    }
+
+    @Test
+    fun `should return the queue AmazonSQS client`() {
+      assertThat(hmppsQueues[0].sqsClient).isEqualTo(sqsClient)
+    }
+
+    @Test
+    fun `should subscribe to the topic`() {
+      verify(snsClient).subscribe(
+        check { subscribeRequest ->
+          assertThat(subscribeRequest.topicArn).isEqualTo("some topic arn")
+          assertThat(subscribeRequest.protocol).isEqualTo("sqs")
+          assertThat(subscribeRequest.endpoint).isEqualTo("http://localhost:4566/queue/some-queue-name")
+          assertThat(subscribeRequest.attributes["FilterPolicy"]).isEqualTo("some topic filter")
+        }
+      )
+    }
+  }
+
+  @Nested
+  inner class `Create AWS HmppsQueue with topic subscription` {
+    private val someQueueConfig = QueueConfig(subscribeTopicId = "sometopicid", subscribeFilter = "some topic filter", queueName = "some queue name", queueAccessKeyId = "some access key id", queueSecretAccessKey = "some secret access key")
+    private val someTopicConfig = TopicConfig(arn = "some topic arn", accessKeyId = "topic access key", secretAccessKey = "topic secret")
+    private val hmppsSqsProperties = HmppsSqsProperties(queues = mapOf("somequeueid" to someQueueConfig), topics = mapOf("sometopicid" to someTopicConfig))
+    private val sqsClient = mock<AmazonSQS>()
+    private val snsClient = mock<AmazonSNS>()
+    private val topics = listOf(HmppsTopic(id = "sometopicid", arn = "some topic arn", snsClient = snsClient))
+    private lateinit var hmppsQueues: List<HmppsQueue>
+
+    @BeforeEach
+    fun `configure mocks and register queues`() {
+      whenever(sqsFactory.awsSqsClient(anyString(), anyString(), anyString(), anyString(), anyString(), anyBoolean()))
+        .thenReturn(sqsClient)
+      whenever(sqsClient.getQueueUrl(anyString())).thenReturn(GetQueueUrlResult().withQueueUrl("some queue url"))
+
+      hmppsQueues = hmppsQueueFactory.createHmppsQueues(hmppsSqsProperties, topics)
+    }
+
+    @Test
+    fun `should return the queue AmazonSQS client`() {
+      assertThat(hmppsQueues[0].sqsClient).isEqualTo(sqsClient)
+    }
+
+    @Test
+    fun `should not subscribe to the topic`() {
+      verifyNoMoreInteractions(snsClient)
+    }
```

Comment:
I'm not sure I understand this test - why doesn't the queue subscribe to the topic?

Reply:
subscription is for localstack and this test is for provider=aws.
This is the same test as HmppsQueueFactoryTest but without the dlq